cnf-certification-test/lifecycle/suite.go

```diff
@@ -36,19 +37,103 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 		provider.BuildTestEnvironment()
 		env = provider.GetTestEnvironment()
 	})
+	testContainersPreStop(&env)
+	testContainersImagePolicy(&env)
+	testContainersReadinessProbe(&env)
+	testContainersLivenessProbe(&env)
+	testPodsOwnerReference(&env)
+})
+
+func testContainersPreStop(env *provider.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestShudtownIdentifier)
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		gomega.Expect(true).To(gomega.Equal(true))
 		badcontainers := []string{}
 		for _, cut := range env.Containers {
-			fmt.Println("container ", cut.Data.Name)
-			logrus.Debugln("check container ", cut.Data.Name) // maybe use different platform ?
+			logrus.Debugln("check container ", cut.Namespace, " ", cut.Podname, " ", cut.Data.Name, " pre stop lifecycle ")
 			if cut.Data.Lifecycle.PreStop == nil {
 				badcontainers = append(badcontainers, cut.Data.Name)
 				logrus.Errorln("container ", cut.Data.Name, " does not have preStop defined")
 			}
 		}
+		if len(badcontainers) > 0 {
+			tnf.ClaimFilePrintf("bad containers %v", badcontainers)
+		}
 		gomega.Expect(0).To(gomega.Equal(len(badcontainers)))
 	})
+}
 
-})
+func testContainersImagePolicy(env *provider.TestEnvironment) {
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestImagePullPolicyIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		badcontainers := []string{}
+		for _, cut := range env.Containers {
+			logrus.Debugln("check container ", cut.Namespace, " ", cut.Podname, " ", cut.Data.Name, " pull policy, should be ", v1.PullIfNotPresent)
+			if cut.Data.ImagePullPolicy != v1.PullIfNotPresent {
+				badcontainers = append(badcontainers, cut.Data.Name)
+				logrus.Errorln("container ", cut.Data.Name, " is using ", cut.Data.ImagePullPolicy, " as image policy")
```

Comment:
Same here. When reviewing the claim file, the customer will only see a list of containers name (that are usually the same across pods) but no reason why they failed the TC, so I'll switch this to ClaimFilePrintf. I prefer this way because customers normally check the DCI's job web results first, which are built by parsing the claim file. The console log should be the last resource for debugging. Also, add pod name and namespace in the trace, please.

Reply:
we are logging all the failed pods in the claimfile so 
```
if(len (badcontainers) > 0{
   tnf...
}
```
should we write that information twice ?